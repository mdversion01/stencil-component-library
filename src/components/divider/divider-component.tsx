// src/components/divider/divider-component.tsx
import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'divider-component',
  styleUrl: 'divider.scss',
  shadow: false,
})
export class DividerComponent {
  @Prop() dashed: boolean = false;
  @Prop() plain: boolean = false;
  @Prop() orientation?: 'left' | 'right' | 'center';
  @Prop() orientationMargin?: string;
  @Prop() type: 'horizontal' | 'vertical' = 'horizontal';
  @Prop() styles?: string;

  private getTextStyle(): { [key: string]: string } | undefined {
    if (!this.styles) return undefined;

    return this.styles.split(';').reduce((acc, pair) => {
      const [key, value] = pair.split(':');
      if (key && value) acc[key.trim()] = value.trim();
      return acc;
    }, {} as { [key: string]: string });
  }


  private getClassNames(base: string) {
    return [
      'divider',
      base,
      this.dashed && 'divider-dashed',
      this.plain && 'divider-plain',
      this.orientation === 'left' && 'divider-with-text-left',
      this.orientation === 'right' && 'divider-with-text-right',
      this.orientation === 'center' && 'divider-with-text-center',
      this.orientation && 'divider-with-text',
      this.orientationMargin === 'left' && 'divider-no-default-orientation-margin-left',
      this.orientationMargin === 'right' && 'divider-no-default-orientation-margin-right',
    ]
      .filter(Boolean)
      .join(' ');
  }

  private renderWithText() {
    return (
      <div
        class={this.getClassNames('divider-horizontal')}
        role="separator"
      >
        <span class="divider-inner-text" style={this.getTextStyle()}>
          <slot></slot>
        </span>
      </div>
    );
  }

  private renderHorizontal() {
    return (
      <div
        class={this.getClassNames('divider-horizontal')}
        role="separator"
      ></div>
    );
  }

  private renderVertical() {
    return (
      <div
        class="divider divider-vertical"
        role="separator"
      ></div>
    );
  }

  render() {
    if (this.orientation === 'left' || this.orientation === 'right' || this.orientation === 'center') {
      return this.renderWithText();
    }

    if (this.type === 'vertical') {
      return this.renderVertical();
    }

    return this.renderHorizontal();
  }
}
