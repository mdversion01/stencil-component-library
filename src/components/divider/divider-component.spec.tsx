// src/components/divider/divider-component.spec.tsx
import { newSpecPage } from '@stencil/core/testing';
import { DividerComponent } from './divider-component';

describe('divider-component', () => {
  it('renders horizontal divider by default', async () => {
    const page = await newSpecPage({
      components: [DividerComponent],
      html: `<divider-component></divider-component>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders dashed horizontal divider', async () => {
    const page = await newSpecPage({
      components: [DividerComponent],
      html: `<divider-component dashed></divider-component>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders vertical divider', async () => {
    const page = await newSpecPage({
      components: [DividerComponent],
      html: `<divider-component type="vertical"></divider-component>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders divider with centered text', async () => {
    const page = await newSpecPage({
      components: [DividerComponent],
      html: `<divider-component orientation="center">Center</divider-component>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders divider with left aligned text and margin override', async () => {
    const page = await newSpecPage({
      components: [DividerComponent],
      html: `<divider-component orientation="left" orientation-margin="left">Left</divider-component>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders divider with custom styles in text', async () => {
    const page = await newSpecPage({
      components: [DividerComponent],
      html: `<divider-component orientation="center" styles="color: red; font-weight: bold;">Styled</divider-component>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
