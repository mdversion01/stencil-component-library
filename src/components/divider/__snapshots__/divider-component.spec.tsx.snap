// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`divider-component renders dashed horizontal divider 1`] = `
<divider-component dashed="">
  <!---->
  <div class="divider divider-dashed divider-horizontal" role="separator"></div>
</divider-component>
`;

exports[`divider-component renders divider with centered text 1`] = `
<divider-component orientation="center">
  <!---->
  <div class="divider divider-horizontal divider-with-text divider-with-text-center" role="separator">
    <span class="divider-inner-text">
      Center
    </span>
  </div>
</divider-component>
`;

exports[`divider-component renders divider with custom styles in text 1`] = `
<divider-component orientation="center" styles="color: red; font-weight: bold;">
  <!---->
  <div class="divider divider-horizontal divider-with-text divider-with-text-center" role="separator">
    <span class="divider-inner-text" style="color: red; font-weight: bold;">
      Styled
    </span>
  </div>
</divider-component>
`;

exports[`divider-component renders divider with left aligned text and margin override 1`] = `
<divider-component orientation="left" orientation-margin="left">
  <!---->
  <div class="divider divider-horizontal divider-no-default-orientation-margin-left divider-with-text divider-with-text-left" role="separator">
    <span class="divider-inner-text">
      Left
    </span>
  </div>
</divider-component>
`;

exports[`divider-component renders horizontal divider by default 1`] = `
<divider-component>
  <!---->
  <div class="divider divider-horizontal" role="separator"></div>
</divider-component>
`;

exports[`divider-component renders vertical divider 1`] = `
<divider-component type="vertical">
  <!---->
  <div class="divider divider-vertical" role="separator"></div>
</divider-component>
`;
